server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,590 @@
+    prefix_dir = Path(hier.path, controller, ".prefix")
+    if not prefix_dir.exists():
+        return 0
+    if not prefix_dir.is_dir():
+        hier.add_unexpected_controllers(
+            Hierarchy.PREFIX_STATUS_INDICATORS, controller, "prefix_dir"
+        )
+        return 1
+
+    prefixes = prefix_dir.glob("*")
+
+    for prefix_path in prefixes:
+        if not prefix_path.name.startswith("prefix.") and not prefix_path.name.endswith(
+            ".prefix"
+        ):
+            hier.add_unexpected_controllers(
+                Hierarchy.NON_PREFIXES, controller, prefix_path.name
+            )
+        elif prefix_path.name.startswith("prefix."):
+            hier.add_unexpected_controllers(
+                Hierarchy.WRONG_PREFIXES, controller, prefix_path.name
+            )
+
+    return 0
+
+
+# 'hier' is an object of ArchiveHierarchy class imported from hierarchy.py
+def verify_archive(hier):
+
+    controllers = Path(hier.path).glob("*")
+    verify_valid_controllers(hier, controllers)
+
+    # now check each "good" controller and get the tarballs it contains
+    for controller in hier.controllers:
+        direct_entries = glob.iglob(os.path.join(hier.path, controller, "*"))
+        hidden_entries = Path(hier.path, controller).glob(".*")
+        if hidden_entries:
+            for hid_entries_path in hidden_entries:
+                if hid_entries_path.is_file():
+                    hier.add_unexpected_controllers(
+                        Hierarchy.UNEXPECTED_OBJECTS, controller, hid_entries_path.name,
+                    )
+        controller_subdir = list()
+        for item in direct_entries:
+            item_path = Path(item)
+            if item_path.is_dir():
+                controller_subdir.append(item_path.name)
+            elif item_path.is_symlink():
+                try:
+                    item_p = item_path.resolve(strict=True)
+                except FileNotFoundError:
+                    print(
+                        f"{item_path.name}_path, '{item_path}', does not resolve to a real location"
+                    )
+                    return 1
+                symlink_item = f"{item_path.name} -> {item_p}"
+                hier.add_unexpected_controllers(
+                    Hierarchy.UNEXPECTED_SYMLINKS, controller, symlink_item
+                )
+            elif all(
+                [
+                    item_path.is_file(),
+                    not item_path.name.endswith(".tar.xz"),
+                    not item_path.name.endswith(".tar.xz.md5"),
+                ]
+            ):
+                hier.add_unexpected_controllers(
+                    Hierarchy.UNEXPECTED_OBJECTS, controller, item_path.name
+                )
+            elif item_path.is_file() and (
+                item_path.name.endswith(".tar.xz")
+                or item_path.name.endswith(".tar.xz.md5")
+            ):
+                hier.add_tarballs(controller)
+            else:
+                print(
```

Comment:
Nobody will see the output of a `print`: you need to add it as an "impossible" situation to the hierarchy so it can be added to the final report.

Reply:
So, should I create another dict value "Impossible" and whatever impossible condition occur I should record that in that?